Testscripts/Linux/Linux-Kernel-Self-Tests.sh

```diff
@@ -0,0 +1,322 @@
+			LKS_SRCDIR="$HOMEDIR/linux"
+		fi
+		
+		cd $LKS_SRCDIR
+		tag=$(git tag | grep $version | head -1)
+		if [[ $tag != "" ]]; then
+			git checkout -f $tag
+		else
+			LogMsg "kernel version: $kernel_version is not supported. Can't find srouce code"
+			SetTestStateAborted
+			exit 0
+		fi
+		return 0
+		
+	fi
+
+	# The kernel is distro kernel
+	case $DISTRO in
+		ubuntu*)
+		#18.04  16.04 19.04            
+			ubuntu_codename="$(awk '/UBUNTU_CODENAME=/' /etc/os-release | sed 's/UBUNTU_CODENAME=//')"
+			if [[ "$ubuntu_codename" != "" ]]; then
+				src_code_git=$ubuntu_src_git$ubuntu_codename
+				LogMsg "kernel source git: $src_code_git"
+				#cd $HOMEDIR
+				git clone $src_code_git
+				check_exit_status "Clone kernel source code" "exit"
+				LKS_SRCDIR=$HOMEDIR/$ubuntu_codename
+				version=$(echo $kernel_version | sed 's/-azure//')					
+			else
+				LogMsg "Get Ubuntu code name failed"
+				SetTestStateAborted
+				exit 0
+			fi
+
+			cd $LKS_SRCDIR
+			tag=$(git tag | grep $version | head -1)
+			if [[ $tag != "" ]]; then
+				git checkout -f $tag
+			else
+				LogMsg "kernel version: $kernel_version is not supported. Can't find srouce code"
+				SetTestStateAborted
+				exit 0
+			fi
+			;;
+		debian*)
+			#debian 10
+			LogMsg "kernel source: from linux-source package"
+			CheckInstallLockUbuntu
+			install_package "linux-source"
+			cd /usr/src/
+			ls linux-source-*.tar.xz
+			if [ $? -ne 0 ]; then
+				LogMsg "Can't find linux srouce code"
+				SetTestStateAborted
+				exit 0
+			fi
+
+			xz -d linux-source-*.tar.xz
+			tar -xf linux-source-*.tar -C $HOMEDIR
+			LKS_SRCDIR="$HOMEDIR/linux-source-*"
+			;;
+		centos*|redhat*)
+			version_id=$(cat /etc/os-release | grep "VERSION_ID" | awk -F '"' '{print $2}' | awk -F '.' '{print $1}')
+			LogMsg "kernel source git: https://git.centos.org/git/rpms/kernel.git"
+			LogMsg "kernel source https://git.centos.org/git/centos-git-common.git"
+			cd /root
+			git clone https://git.centos.org/git/rpms/kernel.git
+			git clone https://git.centos.org/git/centos-git-common.git
+			if [[ -d "kernel" && -d "centos-git-common " ]]; then
+				cd kernel
+				git checkout "c$version_id"
+				../centos-git-common/get_sources.sh
+				if [ -d "SOURCES" ]; then
+					cd SOURCES
+					xz -d linux-*.tar.xz
+					tar -xf linux-*.tar
+					LKS_SRCDIR="/root/kernel/SOURCES/linux-*"
+				fi
+			fi
+			;;
+		*)
+			LogMsg "Unknown distro $DISTRO"
+			SetTestStateAborted
+			exit 0
+		;;
+	esac
+}
+
+# TOTAL_TARGETS is assiged a items string list in which items are in $1 but not in $2
+function skip_targets() {
+	target_item=""
+	for _tar in $1; do
+		flag=0
+		for _skip in $2; do
+			if [ "$_tar"x  == "$_skip"x ]; then
+				flag=1
+				break
+			fi
+		done
+		if [ $flag -eq 0 ]; then
+			target_item=$target_item" $_tar"
+		fi
+	done
+	TOTAL_TARGETS=$(echo $target_item | sed -e 's/^[ ]*//g')
+}
+
+# TOTAL_TARGETS is assiged a items string list of the intersection of $1 and $2
+function check_targets() {
```

Comment:
Same question here. Provide the argument description unless you are using meaningful variables.